### PR TITLE
[crypto] PSA crypto PAL improvements

### PR DESCRIPTION
**Pull request by ATmobica**
Improve usage of external mbedtls in the crypto module.

PSA crypto PAL improvements:
- Use the correct type to refer to the key id.
- Fix AES CCM encrypt/decrypt function - null arguments validation.
- Remove mbedtls PSA crypto condition - it isn't valid for external PSA crypto sources such as TF-M.

### CODE REVIEW

File: src/crypto/crypto.gni [hunk 1]
```diff
@@ -13,14 +13,10 @@
...
+  # Crypto implementation: mbedtls, psa, openssl, boringssl, platform.
   chip_crypto = ""
   chip_with_se05x = 0
 
-  # Compile mbedtls externally. Only used if chip_crypto == "mbedtls"
+  # Compile mbedtls externally.
   chip_external_mbedtls = false
 }
-
-assert(
```
Comment (Damian-Nordic) [inline src/crypto/crypto.gni:24]: Does `chip_external_mbedtls` make sense for other crypto backend than mbedtls or psa?

File: src/crypto/CHIPCryptoPALPSA.cpp [hunk 2]
```diff
@@ -130,17 +126,31 @@ CHIP_ERROR AES_CCM_encrypt(const uint8_t * plaintext, size_t plaintext_length, c
     status = psa_aead_set_nonce(&operation, nonce, nonce_length);
     VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
 
-    status = psa_aead_update_ad(&operation, aad, aad_length);
-    VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
+    if (aad != nullptr && aad_length != 0)
```
Comment (Damian-Nordic) [inline src/crypto/CHIPCryptoPALPSA.cpp:129]:
```suggestion
    if (aad_length != 0)
```
Pointers are already validated if the corresponding lengths are non-zero.

File: src/crypto/CHIPCryptoPALPSA.cpp [hunk 3]
```diff
@@ -130,17 +126,31 @@ CHIP_ERROR AES_CCM_encrypt(const uint8_t * plaintext, size_t plaintext_length, c
...
+    }
+    else
+    {
+        ChipLogDetail(Crypto, "AES_CCM_encrypt: Using aad == null path");
+    }
 
-    status = psa_aead_update(&operation, plaintext, plaintext_length, ciphertext,
-                             PSA_AEAD_UPDATE_OUTPUT_SIZE(PSA_KEY_TYPE_AES, algorithm, plaintext_length), &out_length);
-    VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
+    if (ciphertext != nullptr)
```
Comment (Damian-Nordic) [inline src/crypto/CHIPCryptoPALPSA.cpp:139]:
```suggestion
    if (plaintext_length != 0)
```

File: src/crypto/CHIPCryptoPALPSA.cpp [hunk 4]
```diff
@@ -169,17 +179,32 @@ CHIP_ERROR AES_CCM_decrypt(const uint8_t * ciphertext, size_t ciphertext_length,
     status = psa_aead_set_nonce(&operation, nonce, nonce_length);
     VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
 
-    status = psa_aead_update_ad(&operation, aad, aad_length);
-    VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
+    if (aad != nullptr && aad_length != 0)
```
Comment (Damian-Nordic) [inline src/crypto/CHIPCryptoPALPSA.cpp:182]:
```suggestion
    if (aad_length != 0)
```

File: src/crypto/CHIPCryptoPALPSA.cpp [hunk 5]
```diff
@@ -169,17 +179,32 @@ CHIP_ERROR AES_CCM_decrypt(const uint8_t * ciphertext, size_t ciphertext_length,
...
+    }
+    else
+    {
+        ChipLogDetail(Crypto, "AES_CCM_decrypt: Using aad == null path");
+    }
 
-    status = psa_aead_update(&operation, ciphertext, ciphertext_length, plaintext,
-                             PSA_AEAD_UPDATE_OUTPUT_SIZE(PSA_KEY_TYPE_AES, algorithm, ciphertext_length), &outLength);
-    VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
+    if (ciphertext != nullptr && ciphertext_length != 0)
```
Comment (Damian-Nordic) [inline src/crypto/CHIPCryptoPALPSA.cpp:192]:
```suggestion
    if (ciphertext_length != 0)
```

File: src/crypto/CHIPCryptoPALPSA.cpp [hunk 6]
```diff
@@ -130,17 +126,31 @@ CHIP_ERROR AES_CCM_encrypt(const uint8_t * plaintext, size_t plaintext_length, c
...
+        ciphertext += out_length;
 
-    status = psa_aead_finish(&operation, ciphertext, PSA_AEAD_FINISH_OUTPUT_SIZE(PSA_KEY_TYPE_AES, algorithm), &out_length, tag,
-                             tag_length, &tag_out_length);
+        status = psa_aead_finish(&operation, ciphertext, PSA_AEAD_FINISH_OUTPUT_SIZE(PSA_KEY_TYPE_AES, algorithm), &out_length, tag,
+                                 tag_length, &tag_out_length);
+    }
+    else
+    {
+        status = psa_aead_finish(&operation, ciphertext, 0, &out_length, tag, tag_length, &tag_out_length);
```
Comment (Damian-Nordic) [inline src/crypto/CHIPCryptoPALPSA.cpp:152]:
```suggestion
        status = psa_aead_finish(&operation, nullptr, 0, &out_length, tag, tag_length, &tag_out_length);
```

**Review by Damian-Nordic — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**